lib/pbench/server/api/resources/upload_api.py

```diff
@@ -170,12 +170,17 @@ def _put(self, args: ApiParams, request: Request, context: ApiContext) -> Respon
                     f"File extension not supported, must be {Dataset.TARBALL_SUFFIX}",
                 )
 
-            md5sum = request.headers.get("Content-MD5")
-            if not md5sum:
+            try:
+                md5sum = request.headers["Content-MD5"]
+                if not md5sum:
+                    raise CleanupTime(
+                        HTTPStatus.BAD_REQUEST,
+                        "Missing required 'Content-MD5' header value",
+                    )
+            except KeyError:
                 raise CleanupTime(
                     HTTPStatus.BAD_REQUEST, "Missing required 'Content-MD5' header"
                 )
```

Comment:
This structure is slightly odd:  you're going to some trouble to explicitly raise an exception inside a `try` block which will not catch it.  It would be better if you moved lines 175-179 outside the `try` block at line 173.

Reply:
Fixed.